### PR TITLE
consolidate filter/facet/sorting getstate logic for vertical search

### PR DESCRIPTION
**Pull request by oshi97**
TEST=manual

### CODE REVIEW

File: src/core/core.js [hunk 1]
```diff
@@ -111,16 +109,33 @@ export default class Core {
...
+    // If no verticalKey is set try to get verticalKey from the common search config
+    if (!verticalKey) {
+      verticalKey = this.globalStorage.getState(StorageKeys.SEARCH_CONFIG).verticalKey;
+    }
+
+    // Get filters and facet to send in the request
+    const allFilters = this.globalStorage.getAll(StorageKeys.FILTER);
+    const totalFilter = allFilters.length > 1
+      ? Filter.and(...allFilters)
+      : allFilters[0];
```
Comment (tmeyer2115) [inline src/core/core.js]: Is this safe, what if allFilters is empty?

File: src/core/core.js [hunk 2]
```diff
@@ -280,7 +288,7 @@ export default class Core {
         direction: option.direction
       };
     });
-    this.globalStorage.set(StorageKeys.SORT_BYS, JSON.stringify(sortBys));
+    this.globalStorage.set(StorageKeys.SORT_BYS, sortBys || []);
```
Comment (tmeyer2115) [inline src/core/core.js]: Should we just not call globalStorage.set if sortBys is empty?

File: src/ui/components/filters/filterboxcomponent.js [hunk 3]
```diff
@@ -285,21 +285,11 @@ export default class FilterBoxComponent extends Component {
...
-
-    const facetFilter = this.core.globalStorage.getAll(StorageKeys.FACET_FILTER)[0];
-
     this.core.persistentStorage.delete(StorageKeys.SEARCH_OFFSET);
     this.core.globalStorage.delete(StorageKeys.SEARCH_OFFSET);
     this.core.verticalSearch(this._verticalKey, {
-      input: query,
-      filter: JSON.stringify(totalFilter),
-      facetFilter: JSON.stringify(facetFilter)
+      input: query
```
Comment (tmeyer2115) [inline src/ui/components/filters/filterboxcomponent.js]: let's put everything on the previous line now that the other attributes are gone

**Review by tmeyer2115 — COMMENTED** [review]
Were you able to test each component you modified to make sure it still worked as expected? In future commits, I'd suggest adding a bit more detail about what you did, testing, etc. Follow the format here: https://sites.google.com/a/yext.com/engineering/orientation/style/commit-messages

File: src/core/core.js (hunk 2 above)
Comment (oshi97) [reply src/core/core.js]: Yes !

File: src/ui/components/filters/filterboxcomponent.js (hunk 3 above)
Comment (oshi97) [reply src/ui/components/filters/filterboxcomponent.js]: Done

File: src/core/core.js (hunk 1 above)
Comment (oshi97) [reply src/core/core.js]: Updated

**Review by tmeyer2115 — APPROVED** [review]

**Pull request merged**